pkg/apis/servicecatalog/validation/instance.go

```diff
@@ -88,7 +89,9 @@ func validateServiceInstanceSpec(spec *sc.ServiceInstanceSpec, fldPath *field.Pa
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, validateObjectReferences(spec, fldPath)...)
-	allErrs = append(allErrs, validatePlanReference(&spec.PlanReference, fldPath)...)
+	if !spec.UserProvided {
```

Comment:
I am concerned about having such an important property of instances suddenly becoming optional, mostly because we have so many assumptions in our code, and this may lead to bugs.

I didn't see any proposal or implementation suggestion before this PR came about. It would have been helpful before diving into the code. Was there a design discussion and I missed it?

One idea that I'm not sure if it was considered is to have service catalog provide a placeholder class/plan for user provided services. Just so that we don't need a bunch of special cases for when it's okay to have the plan reference empty. Also it would make the code in the cli and other areas more consistent on how to list/query and display things.

I'm a bit confused about if this replaces the user provided service broker? Does it make things possible that weren't previously possible with that broker? Are they complimentary?

To be clear, I'm not totally against this change at all. I just want to have a bit of discussion before we merge, to make sure that we understand how this will work and if it replaces anything.

Reply:
We discussed this at the last f2f. It's meant to replace the ups broker, although the ups broker is really just a toy while this is intended for actual use.

A blank class/plan is an interesting idea though. What other use cases need such a thing, because I agree it's a bit cumbersome to have to many specific cases that don't require a class/plan.

Reply:
Yeah, if you are OK with redoing this to try out using a placeholder class/plan (maybe in another branch to not lose your work) and see if you like it better?